File: fsharp/sum-of-multiples/SumOfMultiples.fs
module SumOfMultiples

let multiples (number: int) (limit: int): int list =
  [ 
    for n in 1 .. limit do 
      if n < limit && n % number = 0 then 
        yield n 
  ]

let sumOfMultiples (numbers: int list) (upperBound: int): int =
  let mapping x = multiples x upperBound 
  numbers |> List.collect mapping |> List.distinct |> List.sum

File: csharp/sum-of-multiples/SumOfMultiples.cs
using System;
using System.Linq;
using System.Collections.Generic;

public static class SumOfMultiples
{
    public static int Sum(IEnumerable<int> multiples, int max)
    {
        return Enumerable
            .Range(1, max-1)
            .Where(iterator => 
                multiples.Any(multiple => iterator % multiple == 0)
            )
            .Sum();
    }
}